sktree/tree/_unsup_criterion.pyx

```diff
@@ -322,14 +454,18 @@ cdef class TwoMeans(UnsupervisedCriterion):
                 )
 
         # first compute mean
-        mean = self.sum_total / n_node_samples
+        mean = self.sum_total / self.weighted_n_node_samples
 
-        # then compute the impurity as the variance
-        impurity = self.sum_of_squares(
+        # then compute the variance of the cluster
+        sig = self.sum_of_squares(
             self.start,
             self.end,
             mean
         ) / self.weighted_n_node_samples
+
+        # simplified equation of maximum log likelihood function at s=0
+        impurity = -2*(n_node_samples*log(1)-log(2*pi*sig)/2)
```

Comment:
See equation 8.

Reply:
Ah I see, I'll reimplement this equation from the appendix